src/main/java/htsjdk/samtools/cram/build/ContainerParser.java

```diff
@@ -42,6 +42,37 @@ public ContainerParser(final SAMFileHeader samFileHeader) {
         this.samFileHeader = samFileHeader;
     }
 
+    /**
+     * Iterate through all of this container's {@link Slice}s to derive a map of reference sequence IDs
+     * to {@link AlignmentSpan}s.  Used to create BAI Indexes.
+     *
+     * @param container the container to iterate through
+     * @param validationStringency stringency for validating records, passed to
+     * {@link Slice#getMultiRefAlignmentSpans(CompressionHeader, ValidationStringency)}
+     * @return the map of map of reference sequence IDs to AlignmentSpans.
+     */
+    public Map<ReferenceContext, AlignmentSpan> getSpans(final Container container, final ValidationStringency validationStringency) {
```

Comment:
It would be nice if we never had to hand out these maps to other consumers (I'd prefer to push index entry creation into container/slice), but since its a new method, shouldn't it be a `Container` method ?

Reply:
done